homeassistant/helpers/entity_platform.py

```diff
@@ -61,6 +59,35 @@
 DATA_ENTITY_PLATFORM = "entity_platform"
 PLATFORM_NOT_READY_BASE_WAIT_TIME = 30  # seconds
 
+DEVICE_INFO_TYPES = {
+    # Order is important or else link types are detected as primary
```

Comment:
Can we try to improve this comment, maybe
```suggestion
    # Device info is categorized by finding the first device info type which has all
    # the keys of the device info. The link device info type must be kept first
    # to make it preferred over primary.
```